### DISCUSSION TITLE
How to get refreshed data in the in-memory vector store?

### DISCUSSION THREAD

**Discussion by lancerkind** [FlowiseAI/Flowise#1776] [Q&A] [answered]
I'm using the in-memory vector store to work out a design before I bother with a persisting store. I have a chat flow that starts with uploading a CSV, embeds into an in-memory vector store, and then gets to the chat agent. (flow image attached)

Assumption that failed: I had thought that every time I talk with the chat agent, the entire flow would execute each time--upload CSV, embed into an empty in-memory vector store, then eventually get to the chat. But it must be persisting as when I change the CSV data, it doesn't seem to automatically upload the new data and create a new embedding in a fresh in-memory store as the chat agent is still chatting about my earlier CSV data.

What's a simple way to clear the in-memory vector store so I can upload and embed into a clean vector store?
Does the file-uploader upload the same or changed file each time?

<img width="1175" alt="Screenshot 2024-02-21 at 09 28 58" src="https://github.com/FlowiseAI/Flowise/assets/4351057/bf19fb1d-d9f0-4d4e-b1ad-3c87a288eb0f">

**Comment by ori-aventure** [comment]
Not exactly sure if this is what you're asking but isn't that what this button is for? So step 1) change csv data, step 2) hit the upsert button indicated in my screenshot, step 3) hit the chat button right next to it and chat away :)
![vector](https://github.com/FlowiseAI/Flowise/assets/152312095/1dca710c-0fcb-4922-b57e-040828687784)

**Reply by lancerkind** [reply]
You might be on to something here. The upsert doesn't mention dropping the old data. I find upsert confusing because I'm not sure when I need to use that button rather than let the flow handle it. Thanks for the tip. I'll experiment a bit and see if that resolves my question.

**Reply by lancerkind** [reply]
It didn't resolve my question. But thanks for the attempt.

**Comment by lancerkind** [comment]
What's interesting is that even if I delete the old file uploader node and replace it with a new file uploader node, select a new file and upload it, the in-memory database still persists with the old data.  

If I delete the old in-memory database node and put in a new one, it still has the in-memory database with the same old data.

What does it mean to delete a flowise node? I'm imagining that changes the javascript code in the background. But then why would it still maintain the previous flow instead of being replaced with my new flow?

**Reply by Jaredude** [reply]
You would need to save the flow each step of the way. Delete the file uploader. Save. Then add the new file uploaded node, select a new file, save, then upsert.

Same with the in-memory node. Delete. Save. Add new in-memory node.

See if that resolves the issue.

**Reply by lancerkind** [reply]
That sounds very plausible. It sounds like the act of "saving" is possibly what triggers code generation and adjusts the code. I'll give that a try and report back. Thank you for the suggestion!

**Comment by lancerkind** [comment]
To reset a chat flow so you can feed it new data (as shown in the original question's picture) you need to:

**Comment by lancerkind** [comment, marked as answer]
To reset a chat flow so you can feed it new data (as shown in the original poster's picture) you need to do the following:

1. make the changes wanted to the data file accessed by the file uploader (see attached picture)
2. click "Clear Chat History" icon and click Save icon
3. delete in-memory vector store and click Save icon
4. add in new in-memory vector store and click Save icon

Then in the next chat, it will check it's cached chat history but it will be blank so then: the new data will flow from the file uploader, be embedded into the in-memory vector store, and the new data in the vector store will be used to pass along the top "winners" from the DB to the LLM and the chat agent will sort out what it wants to generate.

Thank you to @flang-ls1 and @Jaredude  for your suggestions as they led me to the answer.

<img width="240" alt="clear chat history" src="https://github.com/FlowiseAI/Flowise/assets/4351057/077b1e72-d950-43b9-a5f5-2d90147095e9">

**Reply by Jaredude** [reply]
Glad you were able to sort it out!
